content/en/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md

```diff
@@ -0,0 +1,143 @@
+per pod storage and ordered graceful deployment, scaling and rolling updates. You can think of StatefulSet as the atomic building
+block for running complex stateful applications. As the use of Kubernetes has grown, so has the number of scenarios requiring
+StatefulSets. Many of these scenarios, require faster rolling update then the currently supported one pod at a time, in the case
+where you're using the `OrderedReady` Pod management policy for a StatefulSet.
+ 
+Here are some examples:
+
+1. I am using a StatefulSet to orchestrate a multi instance, cache based application where the size of the cache is large. The cache 
+   starts cold and requires some siginificant amount of time before the container can start. There could be more initial startup tasks
+   that are required. A RollingUpdate on this StatefulSet would take a lot of time before the application is fully updated. If the 
+   StatefulSet supported updating more than one pod at a time, it would result in a much faster update.
+
+2. My stateful application is composed of leaders and followers or one writer and multiple readers. I have multiple readers or 
+   followers and my application can tolerate multiple pods going down at the same time. I want to update this application more than
+   one pod at a time so that i get the new updates rolled out quickly, especially if the number of instances of my application are
+   large. Note that my application still requires unique identity per pod.
+
+
+In order to support such scenarios, Kubernetes 1.24 includes a new alpha feature to help. Before you can use the new feature you must
+enable the `MaxUnavailableStatefulSet` feature flag. Once you enable that, you can specify a new field called `maxUnavailable`, part 
+of the `spec` for a StatefulSet. For example: 
+
+```
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: web
+  namespace: default
+spec:
+  podManagementPolicy: OrderedReady  # you must set OrderedReady
+  replicas: 5
+  selector:
+    matchLabels:
+      app: nginx
+    spec:
+      containers:
+      - image: k8s.gcr.io/nginx-slim:0.8
+        imagePullPolicy: IfNotPresent
+        name: nginx
+  updateStrategy:
+    rollingUpdate:
+      maxUnavailable: 2 # this is the new alpha field, whose default value is 1
+      partition: 0
+    type: RollingUpdate
+```
+
+If you enable the new feature and you don't specify a value for `maxUnavailable` in a StatefulSet, Kubernetes applies a default
+`maxUnavailable: 1`. This matches the behavior you would see if you don't enable the new feature.
+
+I'll run through a scenario based on that example manifest to demonstrate how this feature works. I will deploy a 6 replica StatefulSet,
+with `maxUnavailable` set to 2 and `partition` set to 0.
+
+We can trigger a rolling update, by changing the image to `nginx-slim:0.9`. Once we initiate the rolling update, we can watch the pods
```

Comment:
```suggestion
We can trigger a rolling update, by changing the image to `k8s.gcr.io/nginx-slim:0.9`. Once we initiate the rolling update, we can watch the pods
```

we should probably specify the same registry as in the example

Reply:
using a cmd could be also useful 

```bash
kubectl set image statefulset web nginx=k8s.gcr.io/nginx-slim:0.9
```

Reply:
yeah, this imo is not necessary. let me know if any of you feel strongly about it

Reply:
I'm in favor of getting the blog post merged, so long as it renders out and there are no parts that are outright incorrect.

Reply:
yup it is not necessary and is okay without it